services/minespace-web/src/components/dashboard/mine/projectSummaries/ProjectSummariesTable.js

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+import PropTypes from "prop-types";
+import { Table } from "antd";
+import { truncateFilename, dateSorter } from "@common/utils/helpers";
+import { downloadFileFromDocumentManager } from "@common/utils/actionlessNetworkCalls";
+import CustomPropTypes from "@/customPropTypes";
+import { formatDate } from "@/utils/helpers";
+import * as Strings from "@/constants/strings";
+import * as routes from "@/constants/routes";
+import { EDIT_PENCIL } from "@/constants/assets";
+import LinkButton from "@/components/common/LinkButton";
+
+const propTypes = {
+  projectSummaries: PropTypes.arrayOf(CustomPropTypes.variance).isRequired,
+  projectSummaryStatusCodesHash: PropTypes.objectOf(PropTypes.string).isRequired,
+  isLoaded: PropTypes.bool.isRequired,
+};
+
+export class ProjectSummariesTable extends Component {
+
+  transformRowData = (projectSummaries, codeHash, typeHash) =>
+  projectSummaries &&
+  projectSummaries.map((projectSummary) => ({
+      key: projectSummary.project_summary_guid,
+      projectSummary,
+      mine_guid: projectSummary.mine_guid,
```

Comment:
Is this working as intended? It looks like it should be spread

      {key: projectSummary.project_summary_guid,
      ...projectSummary,
      mine_guid: projectSummary.mine_guid, etc}

Reply:
In this case projectSummary was correct due to the pattern I was following with variances. On line 88:98 you can see it is referenced there(dataIndex) for the Edit functionality. There is another way to do it, but it does work fine as is.